assume/markets/clearing_algorithms/redispatch.py

```diff
@@ -37,16 +38,23 @@ class RedispatchMarketRole(MarketRole):
 
     """
 
+    required_fields = ["node"]
+
     def __init__(self, marketconfig: MarketConfig):
         super().__init__(marketconfig)
-
         self.network = pypsa.Network()
         # set snapshots as list from the value marketconfig.producs.count converted to list
         self.network.snapshots = range(marketconfig.market_products[0].count)
-        self.solver = marketconfig.param_dict.get("solver", "glpk")
-        network_path = marketconfig.param_dict.get("network_path")
+        self.network_data = marketconfig.param_dict.get("grid_data")
+        assert self.network_data
+        # set backup marginal cost
+        backup_marginal_cost = marketconfig.param_dict.get("backup_marginal_cost", 10e4)
+        read_pypsa_grid(
```

Comment:
I find this a good approach, but only to be used with buses and lines. The redispatch requires specific handling of power plants to create all those upward and downward power plants. This should stay in here to be able to see how exactly redispatch is handled and modeled. The functions need t be corrected